pyrasterframes/src/test/python/PyRasterFramesTests.py

```diff
@@ -403,6 +409,64 @@ def test_tile_udt_serialization(self):
             long_trip = df.first()["tile"]
             self.assertEqual(long_trip, a_tile)
 
+    def test_udf_on_tile_type_input(self):
+        import numpy.testing
+        # rf_local_add(t, 0) is to force lazy eval; accessing tile.tile is to get at the actual Tile type vs PRT struct
+        df = self.spark.read.rastersource(self.img_uri).withColumn('tile2', rf_local_add_int(col('tile.tile'), 0))
```

Comment:
What can we do to reduce this kind of  friction?

Reply:
Yeh, that sucks..... Feel like we need to have a design discussion about it and all the datasource schemas, what the expectations are, is this a udf only problem, etc.

Reply:
If it's a `udf` only problem, this is where I wonder if a `rf_udf` annotation could be used/constructed/registered that handles the extraction.

Even better would be if we could handle this through automatic Catalyst casts/conversions over Python UDFs but needs more research.

Reply:
Tracked via https://github.com/locationtech/rasterframes/issues/147